mmv1/products/compute/RegionSecurityPolicyRule.yaml

```diff
@@ -0,0 +1,216 @@
+      The name of the security policy this rule belongs to.
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource. Provide this property when you create the resource.
+  - !ruby/object:Api::Type::Integer
+    name: 'priority'
+    description: |
+      An integer indicating the priority of a rule in the list.
+      The priority must be a positive value between 0 and 2147483647.
+      Rules are evaluated from highest to lowest priority where 0 is the highest priority and 2147483647 is the lowest priority.
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'match'
+    description: |
+      A match condition that incoming traffic is evaluated against.
+      If it evaluates to true, the corresponding 'action' is enforced.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'versionedExpr'
+        description: |
+          Preconfigured versioned expression. If this field is specified, config must also be specified.
+          Available preconfigured expressions along with their requirements are: SRC_IPS_V1 - must specify the corresponding srcIpRange field in config.
+        values:
+          - :SRC_IPS_V1
+        default_value: :SRC_IPS_V1
+      - !ruby/object:Api::Type::NestedObject
+        name: 'config'
+        description: |
+          The configuration options available when specifying versionedExpr.
+          This field must be specified if versionedExpr is specified and cannot be specified if versionedExpr is not specified.
+        required: true
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: 'srcIpRanges'
+            description: |
+              CIDR IP address range. Maximum number of srcIpRanges allowed is 10.
+            item_type: Api::Type::String
+  - !ruby/object:Api::Type::String
+    name: 'action'
+    description: |
+      The Action to perform when the rule is matched. The following are the valid actions:
+
+      * allow: allow access to target.
+
+      * deny(STATUS): deny access to target, returns the HTTP response code specified. Valid values for STATUS are 403, 404, and 502.
+
+      * rate_based_ban: limit client traffic to the configured threshold and ban the client if the traffic exceeds the threshold. Configure parameters for this action in RateLimitOptions. Requires rateLimitOptions to be set.
+
+      * redirect: redirect to a different target. This can either be an internal reCAPTCHA redirect, or an external URL-based redirect via a 302 response. Parameters for this action can be configured via redirectOptions. This action is only supported in Global Security Policies of type CLOUD_ARMOR.
+
+      * throttle: limit client traffic to the configured threshold. Configure parameters for this action in rateLimitOptions. Requires rateLimitOptions to be set for this.
+    required: true
```

Comment:
Is `action` required?

Reply:
It is. The Api returns an error 400 with the following msg: 
"message": "Invalid value for field 'resource.action': ''. Invalid action: "